src/confidential-assets/dto/create-confidential-asset.dto.ts

```diff
@@ -0,0 +1,47 @@
+/* istanbul ignore file */
+
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { IsArray, IsOptional, IsString } from 'class-validator';
+
+import { IsDid, IsTicker } from '~/common/decorators/validation';
+import { TransactionBaseDto } from '~/common/dto/transaction-base-dto';
+
+export class CreateConfidentialAssetDto extends TransactionBaseDto {
+  @ApiProperty({
+    description: 'Custom data to be associated with the Confidential Asset',
+    example: 'Some Random Data',
+    type: 'string',
+  })
+  @IsString()
+  readonly data: string;
+
+  @ApiProperty({
+    description: 'The ticker value to be associated with the Confidential Asset',
+    type: 'string',
+    example: 'TICKER',
+  })
+  @IsTicker()
+  @IsOptional()
+  readonly ticker?: string;
+
+  @ApiProperty({
+    description: 'List of auditors for the Confidential Asset',
+    isArray: true,
+    type: 'string',
+    example: ['0xdeadbeef00000000000000000000000000000000000000000000000000000000'],
+  })
+  @IsString()
+  @IsArray()
+  readonly auditors: string[];
+
+  @ApiPropertyOptional({
+    description: 'List of mediators for the Confidential Asset',
+    isArray: true,
+    type: 'string',
+    example: ['0x0600000000000000000000000000000000000000000000000000000000000000'],
+  })
+  @IsOptional()
+  @IsArray()
+  @IsDid()
```

Comment:
```suggestion
  @IsDid({each: true})
```
I thought this was supposed to be used for arrays

Reply:
I'm merging this in and will update once a new CA version is out and have to update the variables